include/zephyr/devicetree.h

```diff
@@ -1896,6 +1896,9 @@
  * The macro "fn" must take one parameter, which will be the node
  * identifier of a child node of "node_id".
  *
+ * The children will be iterated over in the same order as they
+ * appear in the final devicetree.
+ *
```

Comment:
This could still mean depth-first or breadth-first order, right? Do we really want to make guarantees or the order?

Reply:
> This could still mean depth-first or breadth-first order, right?

The collection of child nodes of a given node are not a tree data structure, so "breadth" vs "depth" first doesn't apply here.

> Do we really want to make guarantees or the order?

Yes, this question has come up multiple times from different users.